File: ruby/g21.super/Student.rb
#!/usr/bin/ruby
# class declaration
class Student < Person
  ###### INITIALIZER ######
  def initialize(name, gpa=0.0) # constructor
    super(name)        # super's initializer
    setGPA(gpa)        # call mutator
  end

  ###### ACCESSSORS (SETTERS) ######
  def setGPA (gpa=0.0)
    @gpa = gpa         # set instance data
  end

  ###### ACCESSSORS (GETTERS) ######
  def getGPA
    @gpa               # return instance data
  end
end

File: python/h21.super/Student.py
#!/usr/bin/env python
from Person import Person    # include Person.py

# class declaration
class Student(Person):
  ###### INITIALIZER ######
  def __init__(self, name, gpa=0.0):
    super(Student, self).__init__(name)
    self.gpa = gpa      # call mutator

  ###### PROPERTY DECORATOR ######
  @property
  def gpa(self):
      return self._gpa     # return data

  @gpa.setter
  def gpa (self,gpa):
    self._gpa = gpa      # set data
